### ISSUE TITLE
Advice on forwarder monitoring?

### ISSUE THREAD

**Issue by aegershman** [splunk/docker-splunk#288]
**TL;DR**, is there a `splunk_forwarder_monitor`-type ansible role equivalent?

I'm currently running a deployment composed of 3 `splunk_heavy_forwarder` pods whose only responsibility is to receive logs ([from the Pivotal Cloud Foundry splunk firehose nozzle-- it just emits platform app logs to an HEC](https://github.com/cloudfoundry-community/splunk-firehose-nozzle)), perform props/transform rules, and forward them onto our main corporate indexer cluster.

But it's hard to understand what's happening on those forwarders. How many logs is the cluster forwarding? What does each forwarder's ingestion pipeline look like? etc. To understand these things you have to look at them _individually_ (with something like `kubectl -n splunk port-forward mypod-xyz 8000:8000`).

My dream Splunk experience™ is to have a HECs ingesting & forwarding logs, and a single "forwarder-manager" which aggregates all the data in the monitoring console about the forwarders. That way there's a one-stop solution for admins to look at the state of the forwarder cluster.

[I've found that a `splunk_deployment_server` can work for app bundles, as expected](https://github.com/aegershman/splunk-helm-chart/pull/77/files#diff-3e6281b3e28f805fc9b9b6a5bd681ac6)-- but it doesn't act as a forwarder-monitor.

Does anyone have any ideas? I don't need any search headers, peers, or even a deployment server, honestly-- I just need a way to visualize the status/metrics/etc. of all the HECs in the forwarder cluster.

Thanks for any advice or opinions, I appreciate it!

**Comment by bb03** [comment]
Hi @aegershman, I asked some people about your request and was not able to obtain a clear answer about if something like this exists or is in the roadmap.  I think this is a feature request that would be best brought up to your sales rep.

The alternative would be to build a splunk app that does what you're asking.  The type of data that you're looking for should all be easy to create searches for.  Basically, searching the _internal index for what you will need.  I know it's not much of an answer, and I'll be happy to address any follow up questions you might have.

**Comment by aegershman** [comment]
Hey @bb03 thanks for reaching out. What I'm looking for is this: https://docs.splunk.com/Documentation/Splunk/8.0.0/DMC/Configureforwardermonitoring + https://docs.splunk.com/Documentation/Splunk/8.0.0/DMC/ForwardersDeployment

I'd like to be able to configure using it through `docker-splunk`. Does that make sense? Thanks again 👍

**Comment by bb03** [comment]
@aegershman Yes, it does.  Thanks!

**Comment by aegershman** [comment]
Right on 👌 With that said, do you or anyone know if there's a way to set that up with `docker-splunk`? (or does previous comment still hold?)

**Comment by jmeixensperger** [comment]
Hey @aegershman, I've opened a PR in the splunk-ansible repo (https://github.com/splunk/splunk-ansible/pull/314) that uses 2 new env vars to control forwarder monitoring:
-  DMC_FORWARDER_MONITORING (bool) - if true, creates the saved search which enables forwarder monitoring on all standalone/search head instances. Defaults to False
- DMC_ASSET_INTERVAL (string) - cron schedule for the saved search that is created. Defaults to "3,18,33,48 * * * *" (every 15 mins)

Will these changes meet your requirements for forwarder monitoring?

**Comment by jmeixensperger** [comment]
The proposed change was merged and should be available in the newest 2 releases (7.2.9.1 and 8.0.1). Please re-open if the change doesn't meet your ask.

**Comment by aegershman** [comment]
! apologies for abandoning this. I should've checked this earlier. Thank you, I will try this out ASAP.
